compiler/rustc_mir_build/src/builder/scope.rs

```diff
@@ -890,8 +926,19 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             known_valid_scrutinee: true,
         };
 
+        let valtree = match self.eval_unevaluated_mir_constant_to_valtree(constant) {
+            Ok((valtree, ty)) => {
+                assert!(!ty.has_param());
```

Comment:
what is this assert for exactly?

Reply:
This is an assert that the type is monomorphic.